aten/src/ATen/native/cuda/ForeachBinaryOpList.cu

```diff
@@ -89,7 +89,7 @@ std::vector<Tensor> foreach_tensor_##NAME##_list_kernel_cuda(TensorList tensors1
 
 FOREACH_BINARY_OP_LIST_ALPHA(add, std::plus);
 FOREACH_BINARY_OP_LIST_ALPHA(sub, std::minus);
-FOREACH_BINARY_OP_LIST(mul, std::multiplies);
-FOREACH_BINARY_OP_LIST(div, std::divides);
+FOREACH_BINARY_OP_LIST(mul, std::multiplies, false);
+FOREACH_BINARY_OP_LIST(div, std::divides, true);
```

Comment:
nit: adding a comment in front of these make them more readable: `/*division_op*/false`

There will probably be more binary operations that do int->float promotion in the future, but I suppose we can refactor when they come along